packages/next-auth/src/providers/vk.ts

```diff
@@ -0,0 +1,285 @@
+      country_id?: number
+      city_id?: number
+      city_name?: string
+      from?: number
+      until?: number
+      position?: string
+    }
+    military?: {
+      country_id: number
+      from?: number
+      unit: string
+      unit_id: number
+      until?: number
+    }
+    education?: {
+      university?: number
+      university_name?: string
+      faculty?: number
+      faculty_name?: string
+      graduation?: number
+    }
+    home_town?: string
+    relation?: 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8
+    relation_partner?: {
+      deactivated?: string
+      first_name: string
+      hidden?: number
+      id: number
+      last_name: string
+      can_access_closed?: boolean
+      is_closed?: boolean
+    }
+    personal?: {
+      alcohol?: 1 | 2 | 3 | 4 | 5
+      inspired_by?: string
+      langs?: string[]
+      life_main?: 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8
+      people_main?: 1 | 2 | 3 | 4 | 5 | 6
+      political?: 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9
+      religion?: string
+      smoking?: 1 | 2 | 3 | 4 | 5
+    }
+    universities?: Array<{
+      chair?: number
+      chair_name?: string
+      city?: number
+      country?: number
+      education_form?: string
+      education_status?: string
+      faculty?: number
+      faculty_name?: string
+      graduation?: number
+      id?: number
+      name?: string
+      university_group_id?: number
+    }>
+    schools?: Array<{
+      city?: number
+      class?: string
+      country?: number
+      id?: string
+      name?: string
+      type?: number
+      type_str?: string
+      year_from?: number
+      year_graduated?: number
+      year_to?: number
+      speciality?: string
+    }>
+    relatives?: Array<{
+      id?: number
+      name?: string
+      type: 'parent' | 'child' | 'grandparent' | 'grandchild' | 'sibling'
+    }>
+    counters?: {
+      albums?: number
+      videos?: number
+      audios?: number
+      photos?: number
+      notes?: number
+      friends?: number
+      groups?: number
+      online_friends?: number
+      mutual_friends?: number
+      user_videos?: number
+      followers?: number
+      pages?: number
+    }
+    is_no_index?: 0 | 1
+  }>
+}
+
+export default function VK<
+  P extends Record<string, any> = VkProfile
+>(options: OAuthUserConfig<P>): OAuthConfig<P> {
+  const apiVersion = "5.131" // https://vk.com/dev/versions
+
+  return {
+    id: "vk",
+    name: "VK",
+    type: "oauth",
+    authorization: `https://oauth.vk.com/authorize?scope=email&v=${apiVersion}`,
+    client: {
+      token_endpoint_auth_method: "client_secret_post",
+    },
+    token: `https://oauth.vk.com/access_token?v=${apiVersion}`,
+    userinfo: `https://api.vk.com/method/users.get?fields=photo_100&v=${apiVersion}`,
+    profile(result: P) {
+      const profile = result.response?.[0] ?? {}
+      return {
+        id: profile.id,
+        name: [profile.first_name, profile.last_name].filter(Boolean).join(" "),
+        email: null,
```

Comment:
@ndom91 Maybe email: null is trying to save accounts.email column?

Reply:
No so I meant that the `email` field is not on the `Account` table at all, but the `User` table.

Reply:
@ndom91 So if I understand correctly, the problem is that we're passing the `account` object which can contain all kinds of properties (it's extending from `Record<string, unknown>` and  `Partial<TokenSet>` as seen [here](https://github.com/nextauthjs/next-auth/blob/e498483b23273d1bfc81be68339607f88d411bd6/packages/next-auth/src/core/types.ts#L249)) into the adapter function `linkAccount` which expects only the object of type `Account`.

In this case, I think `Prisma` is working as expected. The proper solution IMO is that we should use the object with the desired type `Account` in the `linkAccount` implementation. Does that make sense?

Reply:
Yeah that makes sense. So basically drop `email`?

Reply:
Let us drop anything but the expected properties... so I worry that another unexpected property could break the adapter again when used with another provider. WDYT?

Reply:
Yeah that makes a lot of sense 👍